src/host/inputBuffer.cpp

```diff
@@ -492,8 +492,7 @@ size_t InputBuffer::Prepend(const std::span<const INPUT_RECORD>& inEvents)
             return STATUS_SUCCESS;
         }
 
-        _vtInputShouldSuppress = true;
-        auto resetVtInputSuppress = wil::scope_exit([&]() { _vtInputShouldSuppress = false; });
+        const auto wakeup = _wakeupReadersOnExit();

```

Comment:
Much cleaner and safer. Also, it's more correct. We would previously call `WakeUpReadersWaitingForData` unconditionally but the storage could still be empty. This error has probably been in conhost since v1.

Reply:
I can't remember 4 years ago why `_vtInputShouldSuppress` and input wake muffling was added...

Reply:
It was due to the "circular" control flow, as `_termInput` was unfortunately set up with the callback (boo!) `_HandleTerminalInputCallback`. Additionally, the `Prepend` was implemented with a series of 2 "Append"s. The first one would process the input and call into `_termInput` which would then invoke `_HandleTerminalInputCallback`. This meant that the event was flagged and the waiters woken up before the code had finished to append the 2nd half. To fix this, the flag was added to make the callback stop waking the waiters up.

This PR solves the "knot" by moving all of the waking logic out of the internal append-function and into the outermost, public functions. Now they have perfect control over when to wake up something and when to not do it.